Add Golioth API Training

szczys opened the pull request:
This PR adds a new training module focused on the Golioth REST API.

Note that the structure of the Developer Training site has now been drastically changes:
* Existing training has been moved into the magtag-training directory and branded as "Zephyr Training" 
  * Future work will update this training to use other boards
* API Training and Zephyr Training added to the top nav bar
  * The left nav bar will update to only show the entries for the selected training

szczys committed:
move magtag training modules to subdirectory

Prepare this documents site to house multiple different training
courses.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
copy Golioth Intro into API Training

Add top-level section for API Training and copy the Golioth Intro from
MagTag Training into the new section.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add skeletal framework for API training

Add categories for the three major parts of this training modules. Some
of these may end up being single pages rather than catrgories.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add Discover Golioth section for API Training

Includes guided setup for nRF7002-DK

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
update front matter and filenames

* Use filenames for page ordering
* Use front matter for metadata (slug and description)

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
use numbers to organize directories

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add framework for Golioth services walkthrough

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add framework for OpenAPI walkthrough

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
update Device Status for out-of-sync

Highlight the Setting Sync indicator that will show "Out of sync" until
the Settings Service keys are set up on th cloud side.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add LightDB Stream information

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add LightDB State information

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add Remote Logging information

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add Settings Service information

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add OpenAPI explanation and authentication

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add Devices query using OpenAPI Docs

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
remove redundant query status page

Device status is returned with the device list.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add LightDB Stream query

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add Learn from Golioth Web Console page

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add Remote Procedure Call (RPC) information

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add Over-the-Air (OTA) update infomation

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add RPC API page

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
set top nav for API Training and Zephyr Training

* Use the top nav for the two main training types
* Customize the sidebars to display currently selected training material

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
remove unused webapp section

The webapp section may be added in the future, but for now we have not
yet developed content for this module.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add API Training summary

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
update frontpage for two training options

Represent the API Trainging and the Zephyr Training on the front page.

Signed-off-by: Mike Szczys <mike@golioth.io>

ChrisGammell commented on website/docusaurus.config.js:
Don't think I don't see you making the default color scheme dark! :-D

szczys replied:
Oh yeah, just wanted to be sure the page looks right :smiley:

ChrisGammell reviewed (approved):
Changes look great! One small fix I will submit before merging (formatting on one of the API pages)

szczys committed:
update formatting for desired outcome/time

Display the desired outcome and time estimate and a list so it doesn't
take up as much space as it had when styled as individual headings

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
update API summary page

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys commented:
> Changes look great! One small fix I will submit before merging (formatting on one of the API pages)

Thanks for catching that. I actually meant to apply that style change to all of the pages and forgot. I'll include it in a commit before merging.
